### ISSUE TITLE
BEiT intermediate fine-tuning on ImageNet-22k

### ISSUE THREAD

**Issue by liuzhuang13** [microsoft/unilm#378]
Model I am using: BEiT

Hello, thanks for this wonderful work and providing the code and models. Specifically, we would like to work with the BEiT models that were intermediate fine-tuned on ImageNet-22k.

I wonder whether you can provide the training configuration for intermediate fine-tuning on ImageNet-22k, e.g., epochs, data augmentation, learning rate, weight decay, batch size, layer_scale, etc. A training script/command to reproduce the given pre-trained models would be much appreciated. Thanks in advance!

**Comment by donglixp** [comment]
Hi @liuzhuang13 , 

Sure, we can provide all the commands and hyperparams. We will add them to the readme today.

**Comment by donglixp** [comment]
@liuzhuang13 https://github.com/microsoft/unilm/blob/master/beit/README.md#example-fine-tuning-beit-on-imagenet-22k

**Comment by liuzhuang13** [comment]
Thanks a lot!

**Comment by liuzhuang13** [comment]
Thanks for the very helpful training commands! I have several questions below

1. It seems there is an small error on this in the ImageNet-22k script. I'm guessing the model should be 
beit_base_patch16_224 in the following script.

```OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=16 run_class_finetuning.py \
    --model beit_base_patch16_224_8k_vocab --data_path $DATA_PATH \
    --nb_classes 21841 --data_set image_folder --disable_eval_during_finetuning \
    --finetune https://unilm.blob.core.windows.net/beit/beit_base_patch16_224_pt22k.pth \
    --output_dir $OUTPUT_DIR --batch_size 256 --lr 3e-3 --update_freq 1 \
    --warmup_epochs 5 --epochs 90 --layer_decay 0.65 --drop_path 0.2 \
    --weight_decay 0.05 --enable_deepspeed --layer_scale_init_value 0.1 --clip_grad 3.0```

2. I think the --layer_scale_init_value above will not be used since the checkpoint's values will be loaded after model init, right?

3. The training recipes for the -B and -L models are quite different, and also between different stage of training (e.g., 22k-fine-tune and 1k fine-tune), specifically on drop-path (different values for same model at different stages), clip-grad (None, 1.0 and 3.0 are used), and total batch size. May I ask are these parameters tuned by a search or mainly set based on observations/experiences?  Do you have any recommendations if I skip the first ssl pretrain stage on 22k, and just do 22k supervised pretrain then fine-tune on 1k?

Thanks a lot!

**Comment by donglixp** [comment]
1. Thanks for pointing out the incorrect name. It has been fixed at https://github.com/microsoft/unilm/commit/8c00fbbe7d4285cbf1dfeac9977a14d263cff71f .

2. Yes, the value was reloaded from the pretrained checkpoint rather than being reinitialized.

3. The arguments were mainly set by common practices:

- `drop_path`: the deeper TRMs usually use larger values.
- `clip_grad`: can be decided by the gradient norm value of a dry run after it became stable.
- `batch size`: was determined the number of GPU cards. You could use 2k/4k/8k to fully utilize all the GPUs. Then the learning rate can be scaled according to sqrt(bsz).
